src/tests/lvm/lvm_ha_controller/lvm_ha_controller_test.go

```diff
@@ -0,0 +1,274 @@
+	Expect(err).To(BeNil())
+	Expect(isEventPresent).To(BeTrue())
+
+	// sleep for 15 seconds and again verify state and events
+	logf.Log.Info("sleep for 15 seconds")
+	time.Sleep(15 * time.Second)
+
+	// verify pvc to be in pending state
+	pvcPhase, err = k8stest.GetPvcStatusPhase(app.GetPvcName(), common.NSDefault)
+	Expect(err).ToNot(HaveOccurred(), "failed to get pvc phase")
+	Expect(pvcPhase).Should(Equal(coreV1.ClaimPending), "pvc phase is not pending")
+
+	// verify pvc pre condition failed events
+	isEventPresent, err = k8stest.WaitForPvcNormalEvent(app.GetPvcName(), common.NSDefault, volumeProvisionErrorMsg)
+	Expect(err).To(BeNil())
+	Expect(isEventPresent).To(BeTrue())
+
+	// Scale up the lvm-controller deployment replica to initial replica + 1
+	err = k8stest.RestoreDeploymentReplicas(lvmControllerName, common.NSOpenEBS(), 120, lvmControllerOrgReplica+1)
+	Expect(err).To(BeNil(), "failed to scale  deployment %s, error: %v", lvmControllerName, err)
+
+	// get the no of replicas in lvm-controller deployment
+	// Scale up the lvm-controller deployment
+	logf.Log.Info("Scale up lvm-controller deployment")
+	_, err = k8sinstall.ScaleLvmControllerViaHelm(lvmControllerOrgReplica + 1)
+	Expect(err).To(BeNil(), "failed to scale deployment %s, error: %v", lvmControllerName, err)
+
+	//verify pvc and pv to be bound
+	volUuid, err := k8stest.VerifyVolumeProvision(app.GetPvcName(), common.NSDefault)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(volUuid).ToNot(BeEmpty())
+
+	// use created PVC which is deployed as part of restore app
+	err = app.RefreshVolumeState()
+	Expect(err).ToNot(HaveOccurred())
+
+	// deploy fio pod with created volume
+	logf.Log.Info("deploy fio pod with created volume")
+	err = app.DeployApplication()
+	Expect(err).To(BeNil(), "failed to deploy app")
+
+	// Get the name of the controller pod replica which is active as master at present
+	lease, err := k8stest.GetLease(e2e_config.LvmEngineLeaseName, common.NSOpenEBS())
+	Expect(err).To(BeNil(), "failed to get lease %s in kube-system namespace", e2e_config.LvmEngineLeaseName)
+	Expect(lease).ToNot(BeNil(), "no lease found")
+
+	// get lvm controller master pod i.e spec.holderIdentity from lease
+	initialHolderIdentity := lease.Spec.HolderIdentity
+	Expect(initialHolderIdentity).ToNot(BeNil(), "no lease HolderIdentity found")
+	logf.Log.Info("Lvm controller", "Initial HolderIdentity", initialHolderIdentity)
+
+	// taint all nodes so that after deleting one of the lvm controller pods, now pod should not scheduled
+	for _, node := range nodesWithoutTaint {
+		err = k8stest.AddNoScheduleTaintOnNode(node)
+		Expect(err).To(BeNil(), "failed to taint node %s", node)
+	}
+
+	// delete lvm controller pod which is holding lease
+	err = k8stest.DeletePod(*initialHolderIdentity, common.NSOpenEBS())
+	Expect(err).ToNot(HaveOccurred(), "fio to delete pod %s", *initialHolderIdentity)
```

Comment:
update message here, it is for lvm controller pod not fio

Reply:
fixed